### PR TITLE
Add some high level docs for the TokenService

### PR DESCRIPTION
**Pull request by clemens-msupply**
Thought I write down the motivating thoughts we had on why use the current auth token setup.

### CODE REVIEW

**Comment by andreievg** [comment]
Thanks. I am wondering about our other overall docs, and how they fit with rust docs, would imagine we just link rust docs from overall docs.

**Review by andreievg — APPROVED** [review]

File: service/src/token.rs [hunk 1]
```diff
@@ -79,6 +79,33 @@ pub struct TokenPair {
...
+/// attack can be active.
+///
+/// Q: Why an auth token? and why not just having one long lived session cookie?
+/// A: Having short lived auth tokens makes it possible to issue these tokens to a 3rd party to act
+/// on behave of a user, e.g. a plugin or an external service could use an auth token to gain
+/// temporary access to an API but is not able to refresh the token indefinitely.
+///
+/// Q: Still why an auth token? the long lived refresh token could be invalidated on the server
+/// site to revoke access.
+/// A: This requires a reliable mechanism to do the token invalidation. This can be hard to do. With
```
Comment (andreievg) [inline service/src/token.rs:106]: Hmm not sure if i follow this. 
Refresh token would still need to be invalidated, and also the third party will still need to be given refresh token

Comment (clemens-msupply) [reply service/src/token.rs:106]: I assumed the 3rd party wouldn't get the refresh token, e.g. a 3rd party would periodically get a new token from the "admin" instead...

**Comment by clemens-msupply** [comment]
> Thanks. I am wondering about our other overall docs, and how they fit with rust docs, would imagine we just link rust docs from overall docs.

Yes was wondering this as well and this would probably should go in some other kind of docs. But I wanted to get it out as long as I was in a writing mood ;)

**Pull request merged**